spec/dockerfile_spec.rb

```diff
@@ -0,0 +1,53 @@
+#! /usr/bin/env ruby
+
+require 'rspec/core'
+require 'json'
+require "#{File.join(File.dirname(__FILE__), 'examples', 'running_cluster.rb')}"
+
+describe 'The docker-compose file works' do
+  def which(cmd)
+    exts = ENV['PATHEXT'] ? ENV['PATHEXT'].split(';') : ['']
+    ENV['PATH'].split(File::PATH_SEPARATOR).each do |path|
+      exts.each { |ext|
+        exe = File.join(path, "#{cmd}#{ext}")
+        return exe if File.executable?(exe) && !File.directory?(exe)
+      }
+    end
+    return nil
+  end
+
+  before(:all) do
+    @test_agent = "puppet_test#{Random.rand(1000)}"
+    @docker = which('docker')
+    @compose = which('docker-compose')
+    @timestamps = []
+    if @compose.nil?
```

Comment:
Ah, I think I see now why you added a `which` here - to verify command existence. Rather than recursing everything in path, you could alternatively just run `docker-compose --help` and check for a 0 exit code.